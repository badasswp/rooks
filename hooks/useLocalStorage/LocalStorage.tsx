import type { JSX } from 'react';
import useLocalStorage from './';

/**
 * LocalStorage component.
 *
 * This component is used to demonstrate the
 * use of the useLocalStorage hook.
 *
 * @returns {JSX.Element} The LocalStorage component.
 */
const LocalStorage = (): JSX.Element => {
  const sample = useLocalStorage('sample');

  return (
    <div>
      <p>Local Storage</p>
      <p data-testid="sample">Your local storage item is {sample}</p>
    </div>
  );
};

export default LocalStorage;
