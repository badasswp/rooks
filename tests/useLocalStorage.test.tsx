import React from 'react';
import '@testing-library/jest-dom';
import { getByTestId, render } from '@testing-library/react';

import LocalStorage from '../hooks/useLocalStorage/LocalStorage';

describe('<UseLocalStorage />', () => {
  it('should render the component with no issues', () => {
    const { container } = render(<LocalStorage />);

    expect(container).toMatchSnapshot();
  });

  it('should display the localStorage item using the useLocalStorage hook', () => {
    const localStorageMock = (() => {
      let store: Record<string, string> = {
        sample: 'Hello World',
      };
      return {
        getItem: jest.fn((key: string) => store[key] || null),
        setItem: jest.fn((key: string, value: string) => {
          store[key] = value;
        }),
        removeItem: jest.fn((key: string) => {
          delete store[key];
        }),
        clear: jest.fn(() => {
          store = {};
        }),
      };
    })();

    Object.defineProperty(window, 'localStorage', {
      value: localStorageMock,
      writable: true,
    });

    const { container } = render(<LocalStorage />);

    expect(getByTestId(container, 'sample').textContent).toBe(
      'Your local storage item is Hello World'
    );
  });
});
