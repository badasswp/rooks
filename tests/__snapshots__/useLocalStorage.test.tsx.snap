// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UseLocalStorage /> should render the component with no issues 1`] = `
<div>
  <div>
    <p>
      Local Storage
    </p>
    <p
      data-testid="sample"
    >
      Your local storage item is 
    </p>
  </div>
</div>
`;
